library/datasort.c

```diff
@@ -323,7 +340,7 @@ static struct datasort_chunk *datasort_add_chunk(struct datasort_cfg *dcfg)
 	static const char tpl_suffix[] = "chunk.XXXXXX";
 
 	assert(dcfg);
-	assert(dcfg->dir);
+	assert(datasort_dir);
```

Comment:
What is the difference between the two?
Can this new directory be used for every operation, i.e. datasort and defrag?
This should really be one directory, which by default equals to current logic, and if special config option is specified, then all this sorting/defrag logic just moves into that dir and continues there.